packages/gpl-core/src/sessionWallet.ts

```diff
@@ -0,0 +1,113 @@
+
+  constructor(
+    wallet: Wallet,
+    connection: anchor.web3.Connection,
+    cluster: Cluster | "localnet",
+  ) {
+    this.provider = new anchor.AnchorProvider(connection, wallet, {
+      preflightCommitment: "confirmed",
+    });
+    this.program = new anchor.Program(
+      gpl_session_idl as anchor.Idl,
+      GPLSESSION_PROGRAMS[cluster] as anchor.web3.PublicKey,
+      this.provider);
+  }
+
+  public async get(sessionAccount: PublicKey) {
+    return this.program.account.sessionToken.fetch(sessionAccount);
+  }
+
+  private async create(sessionSignerKeypair: Keypair, targetProgramPublicKey: PublicKey, ownerPublicKey: PublicKey, topUp: boolean = false, validUntilTimestamp: number | null = null) {
+    const sessionSignerPublicKey = sessionSignerKeypair.publicKey;
+
+    let validUntilBN: BN | null = null;
+    if (validUntilTimestamp !== null) {
+      validUntilBN = new BN(validUntilTimestamp);
+    }
+
+    const instructionMethodBuilder = this.program.methods.createSession(topUp, validUntilBN)
+      .accounts({
+        targetProgram: targetProgramPublicKey,
+        sessionSigner: sessionSignerPublicKey,
+        authority: ownerPublicKey,
+      });
+
+    const pubKeys = await instructionMethodBuilder.pubkeys();
+    const sessionPDA = pubKeys.sessionToken as PublicKey;
+
+    return {
+      instructionMethodBuilder: instructionMethodBuilder,
+      sessionPDA: sessionPDA,
+    };
+  }
+
+  public async createSession(sessionSignerKeypair: Keypair, targetProgramPublicKey: PublicKey, ownerPublicKey: PublicKey, validUntilTimestamp: number | null = null) {
+    try {
+      const { instructionMethodBuilder, sessionPDA } = await this.create(sessionSignerKeypair, targetProgramPublicKey, ownerPublicKey, false, validUntilTimestamp);
+      
+      return {
+        instructionMethodBuilder: instructionMethodBuilder,
+        sessionPDA: sessionPDA,
+      };
+    } catch (error) {
+      console.error('Error creating session:', error);
+      throw new Error('Failed to create session.');
+    }
+  }
+
+  public async revoke(sessionAccount: PublicKey, owner: PublicKey) {
+    try {
+      return this.program.methods.revokeSession()
+        .accounts({
+          sessionToken: sessionAccount,
+          authority: owner,
+        });
+    } catch (error) {
+      console.error('Error revoking session:', error);
+      throw new Error('Failed to revoke session.');
+    }
+  }
+
+  public async signTransaction(transaction: anchor.web3.Transaction, sessionSignerKeypair: Keypair) {
+    try {
+      const sessionSignerWallet = new anchor.Wallet(sessionSignerKeypair);
+      return await sessionSignerWallet.signTransaction(transaction);
+    } catch (error) {
+      console.error('Error signing transaction:', error);
+      throw new Error('Failed to sign transaction.');
+    }
+  }
+
+  public async signAllTransactions(transactions: anchor.web3.Transaction[], sessionSignerKeypair: Keypair) {
```

Comment:
You can remove this function here. Any immediate consequence of removing this that I am missing?

Reply:
removed all the transaction related function in this class